libsolidity/ast/AST.cpp

```diff
@@ -130,18 +130,23 @@ SourceUnitAnnotation& SourceUnit::annotation() const
 std::set<SourceUnit const*> SourceUnit::referencedSourceUnits(bool _recurse, std::set<SourceUnit const*> _skipList) const
 {
 	std::set<SourceUnit const*> sourceUnits;
+	referencedSourceUnits(sourceUnits, _recurse, _skipList);
+	return sourceUnits;
+}
+
+void SourceUnit::referencedSourceUnits(std::set<SourceUnit const*>& _referencedSourceUnits, bool _recurse, std::set<SourceUnit const*>& _skipList) const
+{
 	for (ImportDirective const* importDirective: filteredNodes<ImportDirective>(nodes()))
 	{
 		auto const& sourceUnit = importDirective->annotation().sourceUnit;
 		if (!_skipList.count(sourceUnit))
 		{
 			_skipList.insert(sourceUnit);
```

Comment:
If we're optimizing this, we could also save a set lookup here:
```suggestion
		auto [skipListIt, notOnSkipListYet] = _skipList.insert(sourceUnit);
		if (notOnSkipListYet)
		{
```

Reply:
I will include this change.